index.php

```diff
@@ -23,22 +23,22 @@
 
     $data = file_get_contents($_FILES['data']['tmp_name']);
 
-    $file = fopen($thefiledata, 'w');
+    $file = fopen($thefiledata, 'wb');
     fwrite($file, $data);
     fclose($file);
 
-    $filename = fopen($thefilename, 'w');
+    $filename = fopen($thefilename, 'wb');
```

Comment:
Note sure here:  `$thefilename`'s content is a text content, containing the filename, not raw bytes data.